src/bin/pgcopydb/snapshot.c

```diff
@@ -79,6 +73,30 @@ copydb_export_snapshot(TransactionSnapshot *snapshot)
 	bool readOnly = false;
 	bool deferrable = true;
 
+	/*
+	 * If the source server is in recovery mode,
+	 * set the isolation level to REPEATABLE READ and read-only mode.
+	 */
+	bool pg_is_in_recovery = false;
+	if (!pgsql_is_in_recovery(pgsql, &pg_is_in_recovery))
+	{
+		/* errors have already been logged */
+		(void) pgsql_finish(pgsql);
+		return false;
+	}
+
+	if (pg_is_in_recovery)
+	{
+		level = ISOLATION_REPEATABLE_READ;
+		readOnly = true;
+	}
```

Comment:
Do we need actually ISOLATION_SERIALIZABLE for normal operations on a primary server? I mean, it might be that we only ever need ISOLATION_REPEATABLE_READ in all cases?